tests/integration2/security/healthcheck/mtls_healthcheck_test.go

```diff
@@ -0,0 +1,84 @@
+//  Copyright 2019 Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+// Package healthcheck contains a test to support kubernetes app health check when mTLS is turned on.
+// https://github.com/istio/istio/issues/9150.
+package healthcheck
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"istio.io/istio/pkg/test/deployment"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/api/descriptors"
+	"istio.io/istio/pkg/test/framework/api/lifecycle"
+	"istio.io/istio/pkg/test/framework/runtime/components/environment/kube"
+)
+
+func TestMain(m *testing.M) {
+	rt, _ := framework.Run("mtls_healthcheck", m)
+	os.Exit(rt)
+}
+
+// TestMtlsHealthCheck verifies Kubernetes HTTP health check can work when mTLS
+// is enabled.
+
+func TestMtlsHealthCheck(t *testing.T) {
+	ctx := framework.GetContext(t)
+
+	// Test requires this Helm flag to be enabled.
+	kube.RegisterHelmOverrides(map[string]string{
```

Comment:
This is not an ideal interface. Are we comfortable with these tests being k8s only? or do we also want to support native?

For k8s, we probably want to expose this on the k8s environment, itself. 

@smawson is working on a change to support per-component configuration: https://github.com/istio/istio/pull/11466.  I suggest you hold off until that lands and then use that as the way of configuring the k8s environment.

Reply:
Didn't see this comments before. Let me take a look of @smawson 's PR